### ISSUE TITLE
Support --listdeps functionality

### ISSUE THREAD

**Issue by oalders** [houseabsolute/perl-code-tidyall#89]
It would be nice if I could do something like `tidyall --listdeps | xargs cpm install -g`.  This would make it easier to stay caught up on dependencies after something changes in `tidyall.ini`.  Leaving this here as a wishlist item.

**Comment by autarch** [comment]
Yeah, that sounds like a great feature. Patches definitely welcome.

**Comment by oalders** [comment]
Where would this code live? In `lib/Code/TidyAll.pm` and get called from `bin/tidyall`?

**Comment by oalders** [comment]
I've been thinking about this.  I think it would be helpful to have something like `--list-perl-deps`, `--list-node-deps` and maybe `--list-apt-deps`.  Ideally you could take the deps and pipe them into `cpanm` or `apt-get` so that your Travis build keeps up with your `tidyall` config.  Thoughts on this (and names)?  There could be some overlap between the apt and node deps, but the main thing I'd like to solve is that just installing Perl deps won't necessarily be enough if you've got non-Perl deps to deal with.

**Comment by autarch** [comment]
That all sounds good too, but I think for now it'd make sense to just focus on the Perl deps. That's a relatively easy problem to solve. The others are a bit more complicated and require support from the each individual plugin to be useful.

**Comment by autarch** [comment]
To clarify, I think that other stuff should be a separate issue.

**Comment by oalders** [comment]
Sure, that makes sense.  I'll start with the Perl deps and keep it simple.

**Comment by autarch** [comment]
I started working on this, as well as moving the contents of the tidyall script to a module, in a branch https://github.com/houseabsolute/perl-code-tidyall/tree/autarch/refactor-cli
